ui/app/pages/confirmation/confirmation.js

```diff
@@ -0,0 +1,239 @@
+  const originMetadata = useOriginMetadata(pendingConfirmation?.origin);
+  const [alertState, dismissAlert] = useAlertState(pendingConfirmation);
+
+  // Generating templatedValues is potentially expensive, and if done on every render
+  // will result in a new object. Avoiding calling this generation unnecessarily will
+  // improve performance and prevent unnecessary draws.
+  const templatedValues = useMemo(() => {
+    return pendingConfirmation
+      ? getTemplateValues(pendingConfirmation, t, dispatch)
+      : {};
+  }, [pendingConfirmation, t, dispatch]);
+
+  useEffect(() => {
+    const environmentType = getEnvironmentType();
+    // If the number of pending confirmations reduces to zero when the user
+    // is in the fullscreen or popup UI, return them to the default route.
+    // Otherwise, if the number of pending confirmations reduces to a number
+    // that is less than the currently viewed index, reset the index.
+    if (
+      pendingConfirmations.length === 0 &&
+      (environmentType === ENVIRONMENT_TYPE_FULLSCREEN ||
+        environmentType === ENVIRONMENT_TYPE_POPUP)
+    ) {
+      history.push(DEFAULT_ROUTE);
+    } else if (pendingConfirmations.length <= currentPendingConfirmation) {
+      setCurrentPendingConfirmation(pendingConfirmations.length - 1);
+    }
+  }, [pendingConfirmations, history, currentPendingConfirmation]);
+  if (!pendingConfirmation) {
+    return null;
+  }
+
+  return (
+    <div className="confirmation-page">
+      {pendingConfirmations.length > 1 && (
+        <div className="confirmation-page__navigation">
+          <p>
+            {currentPendingConfirmation + 1} of {pendingConfirmations.length}{' '}
+            pending
+          </p>
+          {currentPendingConfirmation > 0 && (
+            <button
+              className="confirmation-page__navigation-button"
+              onClick={() =>
+                setCurrentPendingConfirmation(currentPendingConfirmation - 1)
+              }
+            >
+              <i className="fas fa-chevron-left"></i>
+            </button>
+          )}
+          <button
+            className="confirmation-page__navigation-button"
+            disabled={
+              currentPendingConfirmation + 1 === pendingConfirmations.length
+            }
+            onClick={() =>
+              setCurrentPendingConfirmation(currentPendingConfirmation + 1)
+            }
+          >
+            <i className="fas fa-chevron-right"></i>
+          </button>
+        </div>
+      )}
+      <div className="confirmation-page__content">
+        <Box justifyContent="center">
+          <NetworkDisplay
+            colored={false}
+            indicatorSize={SIZES.XS}
+            labelProps={{ color: COLORS.BLACK }}
+          />
+        </Box>
+        <Box justifyContent="center" padding={[1, 4, 4]}>
+          <Chip
+            label={stripHttpsScheme(originMetadata.origin)}
+            labelColor="gray"
```

Comment:
I don't see a `labelColor` prop :thinking: Was this meant to be `borderColor`, or a `labelProp` perhaps?